AK/Math.h

```diff
@@ -304,7 +305,7 @@ CONSTEXPR T log2(T x)
 template<Integral T>
 CONSTEXPR T log2(T x)
 {
-    return x ? 8 * sizeof(T) - count_leading_zeroes(static_cast<MakeUnsigned<T>>(x)) : 0;
+    return x ? (8 * sizeof(T) - count_leading_zeroes((AK::Detail::MakeUnsigned<T>)x) - 1) : NumericLimits<T>::min();
```

Comment:
Why did we need to remove the static cast? Seems unrelated?

Reply:
oops